src/transformers/modeling_utils.py

```diff
@@ -1574,7 +1574,15 @@ def _initialize_weights(self, module):
         """
         if getattr(module, "_is_hf_initialized", False):
             return
-        self._init_weights(module)
+
+        if is_deepspeed_zero3_enabled():
+            import deepspeed
```

Comment:
we should not need to check and import deepspeed each time we init a weight IMO. We should probably pass an optional arg for performance WDYT?